icu4c/source/i18n/islamcal.cpp

```diff
@@ -935,9 +952,17 @@ void IslamicUmalquraCalendar::handleComputeFields(int32_t julianDay, UErrorCode
     }
 
     dayOfMonth = (days - monthStart(year, month)) + 1;
+    if (dayOfMonth > INT32_MAX || dayOfMonth < INT32_MIN) {
+        status = U_ILLEGAL_ARGUMENT_ERROR;
+        return;
+    }
```

Comment:
I see three instances in this PR of this pattern, where you're checking some integer value to make sure it's between `INT32_MIN` and `INT32_MAX` and setting the status code to `U_ILLEGAL_ARGUMENT_ERROR` and returning if it's not.  There's also two similar-looking spots at the top of the PR where you're checking `uprv_add32_overflow()` and returning `U_ILLEGAL_ARGUMENT_ERROR`.  It kind of feels like there might be a way to factor these out into a utility function: `FAIL_IF_OVERFLOW(dayOfMonth)` or something like that.  Just to improve readability.  But I'm not going to hold up the PR over that.

Reply:
> FAIL_IF_OVERFLOW(dayOfMonth) 

that will be hard because the variable name of the status is not always the same and the return type of the function are also not the same (some are void and some are int32_t)

Reply:
You're probably right, and it's definitely not a hill I want to die on.